### PR TITLE
Interactive Game Developed Version.

### PR DESCRIPTION
**Pull request by mizanriseup**
This is a pull request to check and merge the code for interactive game with IOGT.

### CODE REVIEW

File: interactive/models.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+from wagtail.models import Page
+from wagtail.admin.panels import FieldPanel
+from home.mixins import PageUtilsMixin, TitleIconMixin
+from django.contrib.auth import get_user_model
+
+from interactive.forms import MessageSendForm
+from interactive.services import ShortCodeService
+
+
+# from interactive.services import ShortCodeService
```
Comment (istride) [inline interactive/models.py:19]: Commented code.

File: interactive/models.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+
+class InteractivePage(Page, PageUtilsMixin, TitleIconMixin):
+    parent_page_types = [
+        "home.HomePage", "home.Section", 'home.FooterIndexPage'
+    ]
+    subpage_types = []
+    template = 'interactive/interactive_game.html'
+
+    # button_text = models.CharField(max_length=255)
```
Comment (istride) [inline interactive/models.py:56]: commented code.

File: interactive/models.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    trigger_string = models.CharField(max_length=255)
+    channel = models.ForeignKey(
+        InteractiveChannel,
+        on_delete=models.PROTECT
+    )
+
+    content_panels = Page.content_panels + [
+        # FieldPanel('button_text'),
```
Comment (istride) [inline interactive/models.py:64]: commented code.

**Review by istride — CHANGES_REQUESTED** [review]
I think this should be cleaned up a bit more to remove redundant comments, commented code and redundant imports. I suggest also that you run the [black](https://black.readthedocs.io/en/stable/index.html) code formatter over this.

File: interactive/admin.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from django.contrib import admin
+
+from django.contrib import admin
```
[the remark this answers is not in the document]
Comment (mizanriseup) [reply interactive/admin.py]: Updated.

File: interactive/api/views.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+class RapidProWebhook(APIView):
+    permission_classes = [IsAuthenticated, IsRapidProGroupUser]
+
+    def post(self, request):
+        serializer = RapidProMessageSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        # Extract validated data from serializer
```
[the remark this answers is not in the document]
Comment (mizanriseup) [reply interactive/api/views.py]: Updated

File: interactive/models.py (hunk 1 above)
Comment (mizanriseup) [reply interactive/models.py]: Updated.

File: interactive/models.py (hunk 2 above)
Comment (mizanriseup) [reply interactive/models.py]: Updated

File: interactive/models.py (hunk 3 above)
Comment (mizanriseup) [reply interactive/models.py]: Updated

**Review by istride — APPROVED** [review]

**Pull request merged**